test/widgets/buttons/toggle_button_test.dart

```diff
@@ -0,0 +1,132 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_screenutil/flutter_screenutil.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:tackleapp/theme/tokens/color_tokens.dart';
+import 'package:tackleapp/widgets/buttons/toggle_button.dart';
+
+import '../constants/key_constants.dart';
+
+Widget buildToggleButton(bool value, Function(bool) onChanged) =>
+    ScreenUtilInit(
+      designSize: const Size(440, 956),
+      builder: (context, child) => MaterialApp(
+        home: Scaffold(
+          body: MinyToggleButton(
+            value: value,
+            onChanged: onChanged,
+          ),
+        ),
+      ),
+    );
+
+void main() {
+  final toggleSliderFinder = find.byKey(toggleSliderFinderKey);
+  final toggleBackgroundFinder = find.byKey(toggleBackgroundFinderKey);
+
+  testWidgets(
+      'Given initial value is false '
+      'When toggle is pressed '
+      'Then alignment changes to right and color is primary',
+      (WidgetTester tester) async {
+    var toggledValue = false;
+
+    await tester.pumpWidget(
+      buildToggleButton(toggledValue, (value) {
+        toggledValue = value;
+      }),
```

Comment:
Add More Detailed Assertions:
 Verify intermediate states during the tests, especially after each interaction. // Add intermediate state checks
   await tester.tap(find.byKey(toggleBackgroundFinderKey));
   await tester.pumpAndSettle();
   buttonSlider = tester.widget<Container>(toggleSliderFinder);
   buttonBackground = tester.widget<AnimatedContainer>(toggleBackgroundFinder);
   expect(toggledValue, isTrue);
   expect(buttonBackground.alignment, Alignment.centerRight);
   expect((buttonSlider.decoration as BoxDecoration).color, ColorTokens.primary);